doc/camera_and_viewport.md

```diff
@@ -0,0 +1,45 @@
+
+One important thing to note about the Camera is that since (unlike the Viewport) it's intended to be dynamic, most camera movements won't immediately happen. Instead, the camera has a configurable speed and is updated on the game loop. If you want to immediately move your camera (like on your first camera setup at game start) you can use the `snap` function. Calling snap during the game can lead to jarring or unnatural camera movements though, so avoid that unless desired (say for a map transition, for example). Carefully check the docs for each method for more details about how it affects the camera movement.
+
+Another important note is that the camera is applied after the viewport, and only to non-HUD components. So screen size here is considering the effective size after Viewport transformations.
+
+There are two types of transformations that the Camera can apply to the Canvas. The first and most complex one is translation. That can be applied by several factors:
+
+ * nothing: by default the camera won't apply any transformation, so it's optional to use it.
+ * relative offset: you can configure this to decide "where the center of the camera should be on the screen". By default it's the top left corner, meaning that the centered coordinate or object will always be on the top left corner of the screen. You can smoothly change the relative offset during gameplay (that can be used to apply a dialogue or item pickup temporary camera transition for example).
+ * moveTo: if you want to ad-hoc move your camera you can use this method; it will smoothly transition the camera to a new position, ignoring follows but respecting relative offset and world bounds. This can be reset by `resetMovement` if used in conjunction to follow so that the followed object starts being considered again.
+ * follow: you can use this method so that your camera continuously "follow" an object (for example, a `PositionComponent`). This is not smooth because the movement of the followed object itself is assumed to already be smooth (i.e. if your character teleport the camera will also immediately teleport).
+ * world bounds: when using follow, you can optionally define the bounds of the world. If that is done, the camera will stop following/moving so that out-of-bounds areas are not shown (as long as the world is bigger than the screen).
+
+Finally the second transformation that the camera applies is scaling. That allows for dynamic zooming, and it's controlled by the `zoom` field. There is no zoom speed, that must be controlled by you when changing. The `zoom` variable is immediately applied.
+
+When dealing with events, it is imperative to convert screen coordinates to world coordinates (or, for some reasons, you might want to do the reverse). The Camera provides two functions, `screenToWorld` and `worldToScreen` to easily convert between these coordinate spaces.
```

Comment:
```suggestion
When dealing with touch/click events, it is imperative to convert screen coordinates to world coordinates (or, for some reasons, you might want to do the reverse). The Camera provides two functions, `screenToWorld` and `worldToScreen` to easily convert between these coordinate spaces.
```

Reply:
not only touch or click, any events. drag, scroll, etc

Reply:
Drag are touch events right? Scroll you don't need to translate I think. Anyway the idea here is to make it clear that those events are input events, I think it wasn't very clear which events the sentence is talking about

Reply:
scrolls also have positions. I will add input